src/commands/fate.js

```diff
@@ -44,84 +66,155 @@ const updateCharacter = (update, character) => ({
     ...update,
 })
 
-const refreshFate = name =>
+const updatePoints = amount =>
+    character =>
+        updateCharacter({ points: character.points + Number(amount) }, character)
+
+const updateRefresh = amount =>
+    character =>
+        updateCharacter({ refresh: Number(amount) }, character)
+
+const refreshPoints = character => {
+    const points = Math.max(character.points, character.refresh)
+    return updateCharacter({ points }, character)
+}
+
+const refreshFate = str =>
     data => {
-        const character = findCharacterOrFail(name)(data)
-        const points = Math.max(character.points, character.refresh)
+        const names = makeCharacterList(str)
+        const characters = findCharactersOrFail(names)(data)
         return data
-            .filter(x => x.name !== name)
-            .concat(updateCharacter({ points }, character))
+            .filter(x => !names.includes(x.name))
+            .concat(characters.map(refreshPoints))
```

Comment:
You're duplicating the pattern:
```javascript
const names = makeCharacterList(str)
const characters = findCharactersOrFail(names)(data)
return data
    .filter(x => !names.includes(x.name))
    .concat(characters.map(someFunc))
```
in `refreshFate`, `setRefresh` and `gainFate` (also in `spendFate`, but that's a special case because it does a bit more).
It's alright as it is, but it feels like it could be abstracted into a higher order function:
```javascript
const updateCharacters = transform => str => data => {
    const names = makeCharacterList(str)
    const characters = findCharactersOrFail(names) (data)
    return data
        .filter(x => !names.includes(x.name))
        .concat(characters.map(transform))
}

const refreshFate = updateCharacters(refreshPoints)
const setRefresh = (refresh, str) => updateCharacters(updateRefresh(refresh)) (str)
const gainFate = (str, increase = 1) => updateCharacters(updatePoints(increase)) (str)

const expectPositiveFate = characters => {
    const illegal = characters.find(({ points }) => points < 0)
    if (illegal != null) throw {
        code: 1,
        message: `${illegal.name} has only ${illegal.points} points.`,
    }
    return characters
}

const spendFate = (str, decrease = 1) => pipe(
    updateCharacters(updatePoints(-decrease)) (str),
    expectPositiveFate
)
  
```